cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggerInfos.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.schedule.trigger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.cdap.api.schedule.Trigger;
+import io.cdap.cdap.api.schedule.TriggerInfo;
+import io.cdap.cdap.api.schedule.TriggeringScheduleInfo;
+import io.cdap.cdap.api.workflow.WorkflowToken;
+import io.cdap.cdap.common.app.RunIds;
+import io.cdap.cdap.internal.app.runtime.ProgramOptionConstants;
+import io.cdap.cdap.internal.app.runtime.schedule.DefaultTriggeringScheduleInfo;
+import io.cdap.cdap.internal.app.runtime.workflow.WorkflowTokenCodec;
+import io.cdap.cdap.spi.events.trigger.mapping.TriggeringPropertyMapping;
+import org.apache.twill.api.RunId;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Helper functions for {@link TriggerInfo}
+ */
+public class TriggerInfos {
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(RunId.class, new RunIds.RunIdCodec())
+    .registerTypeAdapter(TriggerInfo.class, new TriggerInfoCodec())
+    .registerTypeAdapter(WorkflowToken.class, new WorkflowTokenCodec())
+    .create();
+  public static final String TRIGGERING_PROPERTIES_MAPPING = "triggering.properties.mapping";
+
+  /**
+   * Deserialize {@link TriggeringScheduleInfo} if SystemArgs contains the key 'triggeringScheduleInfo'
+   * else returns null
+   * @param sysArgs
+   * @return {@link TriggeringScheduleInfo}
+   */
+  @Nullable
+  public static TriggeringScheduleInfo deserializeTriggeringScheduleInfo(Map<String, String> sysArgs) {
```

Comment:
we should have both this getTriggerScheduleInfo and also refactor things so to add a setTriggerScheduleInfo()  method and have it called here instead of whatever is currently being done. That makes it easier to trace through the code, as you can easily tell who is reading the value that gets set and vice versa..

Reply:
yes it makes sense, i have moved the logic of creating `StartMetadata` to `TriggerInfos`.